axelrod/fingerprint.py

```diff
@@ -246,73 +68,39 @@ def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None,
             produce more Points that will be closer together.
         processes : integer, optional
             The number of processes to be used for parallel processing
+        filename: str, optional
+            Name of output file.
+        in_memory: bool
+            Whether to keep the interactions in memory or use a file.
         progress_bar : bool
             Whether or not to create a progress bar which will be updated
-
         Returns
         ----------
         self.data : dictionary
             A dictionary where the keys are coordinates of the form (x, y) and
             the values are the mean score for the corresponding interactions.
         """
-
-        if on_windows and (filename is None):  # pragma: no cover
-            in_memory = True
-        elif filename is None:
-            outputfile = NamedTemporaryFile(mode='w')
-            filename = outputfile.name
-
-        edges, tourn_players = self.construct_tournament_elements(
-            step, progress_bar=progress_bar)
-
-        self.step = step
-        self.spatial_tournament = axl.SpatialTournament(
-            tourn_players,
-            turns=turns,
-            repetitions=repetitions,
-            edges=edges)
-        self.spatial_tournament.play(build_results=False,
-                                     filename=filename,
-                                     processes=processes,
-                                     in_memory=in_memory,
-                                     progress_bar=progress_bar)
+        tournament_creator = SpatialTournamentCreator(
+            player=self._strategy, probe=self._probe,
+            step=step, progress_bar=progress_bar)
+        tournament_kwargs = {'turns': turns, 'repetitions': repetitions}
+        tournament = tournament_creator.get_tournament(**tournament_kwargs)
+
+        filename, in_memory = update_according_to_os(filename, in_memory)
+        tournament.play(filename=filename, in_memory=in_memory,
+                        processes=processes, progress_bar=progress_bar,
+                        build_results=False)
+
+        point_edge_map = tournament_creator.get_points_to_edges()
         if in_memory:
-            self.interactions = self.spatial_tournament.interactions_dict
+            interactions = tournament.interactions_dict
         else:
```

Comment:
when i run a SpatialTournament (on windows) with the non-windows parameters it writes to automatic file just fine.  it also has the interactions_dict in memory.  is this not the case on linux/mac?

```
>>> x = axl.SpatialTournament(players=[axl.Cooperator(), axl.Cooperator()], edges=[(0, 1)],turns=1, repetitions=1)
>>> x.play(build_results=False,filename=None, in_memory=False)
Playing matches:   0%|          | 0/1 [00:00<?, ?it/s]E:\work\environments\tester\lib\site-packages\axelrod\tournament.py:113: UserWarning: Tournament results will not be accessible since build_results=False and no filename was supplied.
  "Tournament results will not be accessible since "
                                                      
>>> x.interactions_dict 
{(0, 1): [[('C', 'C')]]}
```